src/test/java/core/basesyntax/ReportCreatorServiceTest.java

```diff
@@ -0,0 +1,63 @@
+package core.basesyntax;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.ReportCreatorService;
+import core.basesyntax.service.impl.ReportCreatorServiceImpl;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ReportCreatorServiceTest {
+    private static ReportCreatorService reportCreatorService;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        reportCreatorService = new ReportCreatorServiceImpl();
+    }
+
+    @Test
+    public void createReport_initFruitsList_ok() {
+        List<Fruit> initFruitsList = initFruitList();
+        List<String> actualList = reportCreatorService.createReport(initFruitsList);
+        List<String> expectedList = List.of("fruit,quantity", "apple,20", "banana,30", "orange,55");
+        Assert.assertEquals(
+                String.format("Incorrect result of ReportList for the input:\n%s.", initFruitsList),
+                expectedList, actualList);
+        Assert.assertEquals("Incorrect size of ReportList.Size should be ",
+                expectedList.size(), actualList.size());
+    }
+
+    @Test
+    public void createReport_withoutHeadline_notOk() {
+        List<Fruit> initFruitsList = initFruitList();
+        List<String> actualList = reportCreatorService.createReport(initFruitsList);
+        List<String> expectedList = List.of("fruit,quantity", "apple,20", "banana,30", "orange,55");
+        Assert.assertEquals("Your report should start with a headline: \"fruit,quantity\" ",
+                expectedList.get(0), actualList.get(0));
+    }
+
+    @Test
+    public void createReport_withNullOrEmptyInput_notOk() {
```

Comment:
the same here. Let’s separate these test cases. Also, it’s better to throw exceptions in your main solution.